pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CookiePayload.scala

```diff
@@ -0,0 +1,56 @@
+package com.gu.pandomainauth.service
+
+import com.gu.pandomainauth.service.CookiePayload.encodeBase64
+import org.apache.commons.codec.binary.Base64
+import org.apache.commons.codec.binary.Base64.isBase64
+
+import java.nio.charset.StandardCharsets.UTF_8
+import java.security.{PrivateKey, PublicKey}
+import scala.util.matching.Regex
+
+/**
+ * A representation of the underlying binary data (both payload & signature) in a Panda cookie.
+ *
+ * If an instance has been parsed from a cookie's text value, the existence of the instance
+ * *does not* imply that the signature has been verified. It only means that the cookie text was
```

Comment:
this abstraction seems very useful to me, though I wonder if there's a way to make it more obvious from the name of the case class that having a `CookiePayload` doesn't imply that its signature's been verified?

Reply:
Maybe `UnverifiedCookiePayload`? I think bringing in `Unverified` is a bit tricky though, because `CookiePayload` transforms both ways and so changing its name also affects what the code looks like when we _generate_ a cookie:

https://github.com/guardian/pan-domain-authentication/blob/41c7bafe524a1b2bc4a80c5b78c35eb99bfadc1e/pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CookieUtils.scala#L53

...given that at that point we're literally signing it with a private key, it wouldn't feel quite right to me to be referring to it as an _Unverified_ CookiePayload...

Reply:
ah yes I see your point.. probably it would require having two different data types in that case, which seems like it'd be OTT at this point for a fairly theoretical issue 👍